ui/components/ui/form-field/form-field.stories.js

```diff
@@ -84,3 +88,30 @@ FormFieldWithError.args = {
   titleText: 'Title',
   error: 'Incorrect Format',
 };
+
+export const CustomComponents = (args) => {
+  return (
+    <div style={{ width: '600px' }}>
+      <FormField
+        {...args}
+        TitleTextCustomComponent={
+          <Typography>TitleTextCustomComponent</Typography>
+        }
+        TitleUnitCustomComponent={
+          <Typography marginLeft={2}>TitleUnitCustomComponent</Typography>
+        }
+        TooltipCustomComponent={
+          <Tooltip
+            interactive
+            position="top"
+            html={<Typography>Custom tooltip</Typography>}
+          >
+            <Box as="i" marginLeft={2} className="fa fa-question-circle" />
+          </Tooltip>
+        }
+        titleDetail={<Typography>TitleDetail</Typography>}
+        titleDetailWrapperProps={{ marginBottom: 0 }}
+      />
+    </div>
+  );
```

Comment:
@VSaric example of how this component could work with your use case https://github.com/MetaMask/metamask-extension/pull/15522

Reply:
@georgewrmarshall Wow, this is amazing! Thank you for these updates that could work with my use case! But I have couple of questions. I used this in `form-field.stories.js`:
```
 titleDetail={
          <button style={{
            backgroundColor: clicked
              ? 'var(--color-warning-default)'
              : 'var(--color-background-alternative)',
          }}
          onClick={() => setClicked(!clicked)}
          >
            TitleDetail
          </button>
        }
        titleDetailWrapperProps={{ marginBottom: 0 }}
        customSpendingCapText={<Typography>Only enter a number that you're comfortable with the contract spending now or in the future. You can always increase the spending cap later.</Typography>}
```
I added in `titleDetail` property `button`, not `Typography` like you have here. I added that and `customSpendingCapText` property because of the use case that I must have in #15522. 

My question is, when `titleDetail` property have `button`, why it is clickable everywhere I click? It is maybe a small bug or this scenario is okay? You can see this in the screencast below:


https://user-images.githubusercontent.com/92527393/186387909-aa9e1d50-43cb-4822-b078-d6e74f888fb9.mov

I noticed that all `form-field` component is wrapped around `<label>...</label>` and because of that I think when I have in `titleDetail` property `button`, it is clickable everywhere I click.

Reply:
Ah yes you're absolutely right I've removed the wrapping label component and added it to just the `titleText` here https://github.com/MetaMask/metamask-extension/pull/15679/commits/b0dd98f6ec3ba2523ab5968f2548fd407771b532

Reply:
@georgewrmarshall This is excellent! I posted one more comment for failing unit tests. When tests pass, do you want to open this PR and merge it to the develop? I just wondering because if this PR will go on develop, then I must rebased and wait until this PR was merged to finish my use case in #15522.

Reply:
Yeah apologies this is a blocker. Will work on this today